docs/quickstarts/express.mdx

```diff
@@ -0,0 +1,173 @@
+
+  <CodeBlockTabs options={["npm", "yarn", "pnpm"]}>
+    ```bash {{ filename: 'terminal' }}
+    npm install @clerk/express
+    ```
+
+    ```bash {{ filename: 'terminal' }}
+    yarn add @clerk/express
+    ```
+
+    ```bash {{ filename: 'terminal' }}
+    pnpm add @clerk/express
+    ```
+  </CodeBlockTabs>
+
+  ### Set your Clerk API keys
+
+  <SignedIn>
+    Add the following keys to your `.env.local` file. These keys can always be retrieved from the [API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page of your Clerk Dashboard.
+  </SignedIn>
+
+  <SignedOut>
+    1. Navigate to the Clerk Dashboard.
+    1. In the navigation sidebar, select [API Keys](https://dashboard.clerk.com/last-active?path=api-keys).
+    1. In the **Quick Copy** section, copy your Clerk publishable and secret key.
+    1. Paste your keys into your `.env.local` file.
+
+    The final result should resemble the following:
+  </SignedOut>
+
+  ```sh {{ filename: '.env.local' }}
+  CLERK_PUBLISHABLE_KEY={{pub_key}}
+  CLERK_SECRET_KEY={{secret}}
+  ```
+
+  This guide uses `dotenv` to load the environment variables. Install the package by running the following command:
+
+  <CodeBlockTabs options={["npm", "yarn", "pnpm"]}>
+    ```bash {{ filename: 'terminal' }}
+    npm install dotenv
+    npm install -D @types/dotenv
+    ```
+
+    ```bash {{ filename: 'terminal' }}
+    yarn add dotenv
+    yarn add -D @types/dotenv
+    ```
+
+    ```bash {{ filename: 'terminal' }}
+    pnpm add dotenv
+    pnpm add -D @types/dotenv
+    ```
+  </CodeBlockTabs>
+
+  ### Add middleware to your application
+
+  The [`clerkMiddleware()`](/docs/references/express/overview#clerk-middleware) helper integrates Clerk authentication into your Express application. It can be registered globally or for specific routes. The following example registers it globally.
+
+  ```ts {{ filename: 'index.ts', mark: [3, 8] }}
+  import 'dotenv/config'
+  import express from 'express'
+  import { clerkMiddleware } from '@clerk/express'
+
+  const app = express()
+  const PORT = 3000
+
+  app.use(clerkMiddleware())
+
+  app.listen(PORT, () => {
+    console.log(`Example app listening at http://localhost:${PORT}`)
+  })
+  ```
+
+  ### Accessing auth state using `getAuth()`
+
+  The [`getAuth()`](/docs/references/express/overview#get-auth) helper retrieves authentication state from the request object.
+
+  [`clerkClient`](/docs/references/express/overview#clerk-client) is an instance of Clerk's JavaScript Backend SDK, which provides access to Clerk's Backend API resources and low-level authentication utilities for JavaScript environments.
+
+  The following example uses `getAuth()` to retrieve the `userId`, which is used to protect the route and is passed to `clerkClient.users.getUser()` to retrieve the current user's `User` object.
+
+  > [!NOTE]
+  > If you would rather use middleware to protect routes, check out [`requireAuth`](/docs/references/express/overview#require-auth).
```

Comment:
What are the tradeoffs here? Why would you want to use middleware vs doing it manually? I can't imagine, as a developer, coming to this guide with a pre-baked opinion on whether I wanted to do it like this or use middleware, without knowing what either of them are.

I also kind of feel like the default recommendation should be middleware, and this should be the "if you're trying to do something other than redirecting to sign in if the user isn't signed in and if they are, putting the auth state on the request object, you can do it custom like this" - but I'd bet the majority of people won't need to do it like that

Reply:
requireAuth doesn't redirect to sign in, it returns an error.

the differences are, if you use requireAuth, it will return an error. you can also provide a custom response using error middleware.
and then if you use getAuth, again, you are providing a custom response to the unauthenticated user.

and you're saying, for the quickstart, we should show how to protect a route using requireAuth, and simply use getAuth to get the user's id for something like querying the backend api ?